components/camel-cmis/src/main/java/org/apache/camel/component/cmis/CMISProducer.java

```diff
@@ -57,12 +57,11 @@ public CMISEndpoint getEndpoint() {
     }
 
     public void process(Exchange exchange) throws Exception {
-        CmisObject cmisObject = createNode(exchange);
-        log.debug("Created node with id: {}", cmisObject.getId());
 
-        // copy the header of in message to the out message
-        exchange.getOut().copyFrom(exchange.getIn());
-        exchange.getOut().setBody(cmisObject.getId());
+        CamelCMISActions action = exchange.getIn().getHeader(CamelCMISConstants.CMIS_ACTION, CamelCMISActions.class);
+
+        this.getClass().getDeclaredMethod(action.getMethodName(), Exchange.class).invoke(this , exchange);
```

Comment:
Agree with @oscerd about tests as reflection is actively using.
May be better to use ObjecHelper.invokeMethod or  ObjecHelper.invokeMethodSafe here?

Reply:
I'd use invokeMethod here.

Reply:
I cant build the project and I cant run the existing tests. Maybe if I fix this problem I will write some tests...

Reply:
What is the error you have?

Reply:
It fails on last module. 
Cannot resolve  com.atlassian.jira:jira-rest-java-client-api:jar:5.1.0 , com.atlassian.httpclient:atlassian-httpclient-api:jar:2.0.0-m1 .....
Camel-cmis is successfully built, but if I open CMISProducer.class for example, DefaultProducer, Exchange, Message... it can resolve them, cant find the source. 
![image](https://user-images.githubusercontent.com/15361682/61290315-e2fde300-a7cb-11e9-9e15-1777afe3f1e2.png)

Reply:
You need to have core projects open.

Reply:
How do you mean open ? 
I am opening camel project with all modules, not just camel-cmis

Reply:
It seems you are using IntelliJ Idea. Do CTRL+SHIFT+A and choose "Reimport all maven projects". If this does not help, run in root camel folder "mvn clean install -Pfastinstall" and then reimport.